### PR TITLE
Fix: broken feature

### PR DESCRIPTION
**Pull request by hregis**
@eldy 
User not found when you hide the entity drop-down list on the login page with multicompany and "mc" authentication

### CODE REVIEW

File: htdocs/main.inc.php [hunk 1]
```diff
@@ -570,7 +570,7 @@ function analyseVarsForSqlAndScriptsInjection(&$var, $type)
 			exit;
 		}
 
-		$resultFetchUser=$user->fetch('', $login, '', 1, ($entitytotest > 0 ? $entitytotest : -1));
+		$resultFetchUser=$user->fetch('', $login, '', 1, (!empty($conf->entity) ? $conf->entity : ($entitytotest > 0 ? $entitytotest : -1)));
```
Comment (eldy) [inline htdocs/main.inc.php:573]: Few lines before this, there is a call to checkLoginPassEntity and the entity used into this function must be same than into this $user->fetch
So if you change here, you should change in checkLoginPassEntity too. So best is to change the line 488 $entitytotest	= ...

Comment (hregis) [reply htdocs/main.inc.php:573]: @eldy 
the authentication "mc" defines the "$conf->entity" according to the first "group / entity" to which the user belongs. If I put this change before "checkLoginPassEntity" it does not work

Comment (eldy) [reply htdocs/main.inc.php:573]: So this means the fix introduce a serious security hole.
If the user loaded is not the same than the user we check the authenthication, there is a big problem. 
Parameters of function used to check authentication must be same than parameters used to load the user (we check authentication for an entity $entitytotest line 510 and if authentication is ok, we load a user in another entity (that may be a different user). So we are able with this change to login on a user account using the password of another user.

My understanding is that, when using mc authentication, the parameter $entitytotest is ignored.
If yes, did you try to make into the mc code:
global $entitytotest;
and force also the var $entitytotest to the defined entity
$entitytotest = the new defined entity;
so later when we do the fetch, we make the fetch using the correct one and we keep code for common authentication safe.
Tell me if it works. If not, i may have another idea...

Comment (hregis) [reply htdocs/main.inc.php:573]: @eldy 
yes it works, I just released the 7.0.2 version of the module, thank you

**Pull request closed without merging**